.github/scripts/check_all_skip.py

```
@@ -0,0 +1,293 @@
+            text=True,
+        )
+        if result.returncode == 0:
+            return candidate
+    print(f"::error::base ref {base_ref!r} not found (tried {base_ref!r} and 'origin/{base_ref}')")
+    sys.exit(1)
+
+
+def find_changed_test_files(base_ref: str) -> list[str]:
+    """Return test files (test_*.py) changed between base_ref and HEAD."""
+    # git diff --name-only <base>..HEAD
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)